src/lib/chat/matrix-client.ts

```diff
@@ -74,9 +74,24 @@ export class MatrixClient implements IChatClient {
     return this.matrix.getAccountData(eventType);
   }
 
+  private hasSharedJoinedRoom(userId: string): boolean {
+    const rooms = this.matrix.getRooms();
+
+    return !!rooms.find((room) => {
+      const myMember = room.getMember(this.matrix.getUserId());
+      const targetMember = room.getMember(userId);
+
+      return myMember && myMember.membership === 'join' && targetMember && targetMember.membership === 'join';
+    });
+  }
+
   async getUserPresence(userId: string) {
     await this.waitForConnection();
 
+    if (!this.hasSharedJoinedRoom(userId)) {
```

Comment:
Adding this check will silence errors where we previously attempted to fetch presence data for users without a mutual room connection 'joined / accepted'. By ensuring users share a 'joined' status in a room, we reduce unnecessary `getPresence` calls.

While this mitigates the 'FORBIDDEN' error due to the absence of a shared room, the error can still be thrown for other reasons. However, this check serves as a proactive measure to prevent one of the more common causes.


We can remove this check if we don't mind console errors for unsuccessful presence data fetches due to lack of shared rooms. These are those errors:

<img width="736" alt="Screenshot 2023-10-18 at 21 36 40" src="https://github.com/zer0-os/zOS/assets/39112648/651e723e-d0ef-4a84-9224-df03ea729f71">

Reply:
Yeah, I think  this is a little over complicated just to remove an error from the console. Let's just check for the error code in the catch statement and not log the permission error.

Reply:
Okay yeah sounds good, shall update